doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,428 @@
+  ignored but they separate tokens.
+
+Comments
+  Comments are enclosed between ``(*`` and ``*)``.  They can be
+  nested.  They can contain any character. However, embedded
+  :n:`@string` literals must be correctly closed. Comments are treated
+  as blanks.
+
+Identifiers
+  Identifiers, written :n:`@ident`, are sequences of letters, digits,
+  ``_`` and ``'``, that do not start with a digit or ``'``.  That is,
+  they are recognized by the following grammar (except that the string
+  ``_`` is reserved; it is not a valid identifier):
+
+  .. insertprodn ident subsequent_letter
+
+  .. prodn::
+     ident ::= @first_letter {* @subsequent_letter }
+     first_letter ::= {| a .. z | A .. Z | _ | @unicode_letter }
+     subsequent_letter ::= {| @first_letter | @digit | ' | @unicode_id_part }
+
+  All characters are meaningful. In particular, identifiers are
+  case-sensitive.  :production:`unicode_letter` non-exhaustively
+  includes Latin, Greek, Gothic, Cyrillic, Arabic, Hebrew, Georgian,
+  Hangul, Hiragana and Katakana characters, CJK ideographs,
+  mathematical letter-like symbols and non-breaking
+  space. :production:`unicode_id_part` non-exhaustively includes
+  symbols for prime letters and subscripts.
+
+Numerals
+  Numerals are sequences of digits with an optional fractional part
+  and exponent, optionally preceded by a minus sign. :n:`@int` is an
+  integer; a numeral without fractional or exponent parts. :n:`@num`
+  is a non-negative integer.  Underscores embedded in the digits are
+  ignored, for example ``1_000_000`` is the same as ``1000000``.
+
+  .. insertprodn numeral digit
+
+  .. prodn::
+     numeral ::= {+ @digit } {? . {+ @digit } } {? {| e | E } {? {| + | - } } {+ @digit } }
+     int ::= {? - } {+ @digit }
+     num ::= {+ @digit }
+     digit ::= 0 .. 9
+
+Strings
+  Strings begin and end with ``"`` (double quote).  Use ``""`` to
+  represent a double quote character within a string.  In the grammar,
+  strings are identified with :production:`string`.
+
+Keywords
+  The following character sequences are reserved keywords that cannot
+  be used as identifiers::
+
+    _ Axiom CoFixpoint Definition Fixpoint Hypothesis IF Parameter Prop
+    SProp Set Theorem Type Variable as at by cofix discriminated else
+    end exists exists2 fix for forall fun if in lazymatch let match
+    multimatch return then using where with
+
+  Note that notations and plugins may define additional keywords.
+
+Other tokens
+  The set of tokens defined at any given time can vary because the
+  :cmd:`Notation` command can define new tokens.  A :cmd:`Require`
+  command may load more notation definitions, while the end of a
+  :cmd:`Section` may remove notations.  Some notations are defined in
+  the standard library (see :ref:`thecoqlibrary`) and are generally
+  loaded automatically at startup time.
+
+  Here are the character sequences that Coq directly defines as tokens
+  without using :cmd:`Notation`::
```

Comment:
Aren't some of these defined in the prelude using `Notation`?  For example, <= = ->

Reply:
I thought @jfehrle has done a pass on these symbols to provide an accurate list. To confirm with him.

Reply:
Thanks for your answer! So AFAICT there is nothing to change here. (Although of course, this list will not be kept automatically in sync, contrary to the grammar snippets.)

Reply:
FWIW: unlike the grammar snippets